src/Sylius/Bundle/MetadataBundle/DynamicForm/DynamicFormBuilder.php

```diff
@@ -0,0 +1,116 @@
+
+    /**
+     * @param DynamicFormsChoicesMapInterface $dynamicFormsChildrenMap
+     * @param PropertyAccessorInterface $propertyAccessor
+     */
+    public function __construct(DynamicFormsChoicesMapInterface $dynamicFormsChildrenMap, PropertyAccessorInterface $propertyAccessor)
+    {
+        $this->dynamicFormsChildrenMap = $dynamicFormsChildrenMap;
+        $this->propertyAccessor = $propertyAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildDynamicForm(FormBuilderInterface $builder, $name, $type, array $options = [])
+    {
+        $builder->add($name, $type, $options);
+
+        $this->addModelDataListener($builder, $name, $builder->get($name)->getOption('group'));
+        $this->addSubmittedDataListener($builder, $name);
+    }
+
+    /**
+     * @param FormBuilderInterface $dynamicFormBuilder
+     * @param FormInterface $dynamicForm
+     * @param string $formName
+     */
+    private function addEmbeddedFormField(FormBuilderInterface $dynamicFormBuilder, FormInterface $dynamicForm, $formName)
+    {
+        $embeddedForm = $dynamicFormBuilder->getFormFactory()->createNamed(
+            $dynamicFormBuilder->getName(),
+            $formName,
+            null,
+            ['auto_initialize' => false]
+        );
+
+        $dynamicForm->add($embeddedForm);
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param string $name
+     * @param string $group
+     */
+    private function addModelDataListener(FormBuilderInterface $builder, $name, $group)
+    {
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($builder, $name, $group) {
+            try {
+                $data = $this->propertyAccessor->getValue($event->getData(), $name);
+            } catch (AccessException $exception) {
+                $data = null;
+            }
+
+            if (null === $data || !is_object($data)) {
+                return;
+            }
+
+            $formName = $this->dynamicFormsChildrenMap->getFormNameByGroupAndDataClass($group, get_class($data));
+
+            $options = $event->getForm()->get($name)->get('_form')->getConfig()->getOptions();
+            $event->getForm()->get($name)->add('_form', 'choice', array_merge($options, ['data' => $formName]));
+
+            $this->addEmbeddedFormField($builder->get($name), $event->getForm()->get($name), $formName);
+        });
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param string $name
+     */
+    private function addSubmittedDataListener(FormBuilderInterface $builder, $name)
+    {
+        $builder->get($name)->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) use ($builder, $name) {
+            $formName = $event->getData()['_form'];
+            if (empty($formName)) {
+                return;
+            }
+
+            $this->addEmbeddedFormField($builder->get($name), $event->getForm()->get($name), $formName);
+        });
```

Comment:
If I try to save a metadata without choosing a form I get 2x CSRF errors. 

If I try to choose a twitter form option and then 'deselect' it by choosing the blank dropdown option the last form is not removed.

If I submit the form I get the following exception, am I missing some configuration?

```
[1] Symfony\Component\Form\Exception\OutOfBoundsException: Child "twitter" does not exist.
    at n/a
        in /var/www/petewardSylius/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php line 979

    at Symfony\Component\Form\Form->get('twitter')
        in /var/www/petewardSylius/src/Sylius/Bundle/MetadataBundle/DynamicForm/DynamicFormBuilder.php line 114

    at Sylius\Bundle\MetadataBundle\DynamicForm\DynamicFormBuilder->Sylius\Bundle\MetadataBundle\DynamicForm\{closure}(object(FormEvent), 'form.pre_bind', object(EventDispatcher))
        in  line 

    at call_user_func(object(Closure), object(FormEvent), 'form.pre_bind', object(EventDispatcher))
        in /var/www/petewardSylius/app/cache/dev/classes.php line 4832

    at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(array(object(BindRequestListener), 'preBind'), array(object(TrimListener), 'preSubmit'), array(object(CsrfValidationListener), 'preSubmit'), object(Closure)), 'form.pre_bind', object(FormEvent))
        in /var/www/petewardSylius/app/cache/dev/classes.php line 4765

    at Symfony\Component\EventDispatcher\EventDispatcher->dispatch('form.pre_bind', object(FormEvent))
        in /var/www/petewardSylius/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/ImmutableEventDispatcher.php line 43

    at Symfony\Component\EventDispatcher\ImmutableEventDispatcher->dispatch('form.pre_bind', object(FormEvent))
        in /var/www/petewardSylius/vendor/symfony/symfony/src/Symfony/Component/Form/Form.php line 556
```